nominal/core.py

```diff
@@ -311,6 +339,68 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
         )
 
 
+@dataclass(frozen=True)
+class LogSet:
+    rid: str
+    name: str
+    timestamp_type: LogTimestampType
+    description: str | None
+    _client: NominalClient = field(repr=False)
+
+    def _list_logs_paginated(self) -> Iterable[datasource_logset_api.Log]:
+        request = datasource_logset_api.SearchLogsRequest()
+        while True:
+            response = self._client._logset_client.search_logs(
+                self._client._auth_header,
+                log_set_rid=self.rid,
+                request=request,
+            )
+            yield from response.logs
+            if response.next_page_token is None:
+                break
+            request = datasource_logset_api.SearchLogsRequest(page_token=response.next_page_token)
+
+    def list_logs(self) -> Iterable[Log]:
+        """
+        Get logs within this log set.
+        """
+        # TODO: different name?
```

Comment:
maybe `stream_logs()` is most semantic? Or maybe we just define an `__iter__()` so you can `for log in logset`? (or both)?

Reply:
I like doing both!

Reply:
lets start with `stream_logs()` and if it becomes a common workflow, we can `__iter__` it up

Reply:
Ah ok I added iter but I'll remove